app/src/androidTest/java/org/liberty/android/fantastischmemo/test/functionality/testHistoryFunctions.java

```diff
@@ -0,0 +1,130 @@
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+import java.io.File;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class testHistoryFunctions extends AbstractExistingDBTest {
+
+    private AnyMemoDBOpenHelper newDbHelper;
+    private SQLiteDatabase db;
+    private HistoryDao historyDao;
+
+    public static final String dbPath = "/sdcard/historytestdb.db";
+
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        File newdbFile = new File(dbPath);
+        newdbFile.delete();
+        newDbHelper = AnyMemoDBOpenHelperManager.getHelper(getContext(), dbPath);
+    }
+
+    @Override
+    public void tearDown() throws Exception {
+        super.tearDown();
+        AnyMemoDBOpenHelperManager.releaseHelper(newDbHelper);
+        File newdbFile = new File(dbPath);
+        newdbFile.delete();
+    }
+
+    @Before
+    public void setUpHistoryTableAndDao() {
+        db = newDbHelper.getWritableDatabase();
+        fillHistoryTable(db);
+        historyDao = newDbHelper.getHistoryDao();
+        historyDao.setHelper(newDbHelper);
+    }
+
+    private void fillHistoryTable(SQLiteDatabase db) {
+        Date date = new Date();
+        Long timeStamp = date.getTime();
+        History history1 = new History("/test", 100, timeStamp-10);
+        newDbHelper.insertHistory(history1, db);
+        History history2 = new History("/test", 99, timeStamp-9);
+        newDbHelper.insertHistory(history2, db);
+        History history3 = new History("/test", 88, timeStamp-8);
+        newDbHelper.insertHistory(history3, db);
+        History history4 = new History("/test", 77, timeStamp-7);
+        newDbHelper.insertHistory(history4, db);
+        History history5 = new History("/test", 66, timeStamp-6);
+        newDbHelper.insertHistory(history5, db);
+        History history6 = new History("/test", 55, timeStamp-5);
+        newDbHelper.insertHistory(history6, db);
+        History history7 = new History("/test", 44, timeStamp-4);
+        newDbHelper.insertHistory(history7, db);
+        History history8 = new History("/test", 33, timeStamp-3);
+        newDbHelper.insertHistory(history8, db);
+        History history9 = new History("/test", 22, timeStamp-2);
+        newDbHelper.insertHistory(history9, db);
+        History history10 = new History("/test", 11, timeStamp-1);
+        newDbHelper.insertHistory(history10, db);
+    }
+
+
+    @SmallTest
+    @Test
+    public void testInsertFunction() {
+        int score = 50;
+        List<History> hist = newDbHelper.getHistoryForDB("/test");
+        int count = hist.size();
+        History result = HistoryHelper.addToHistory(score, count, hist, "/test", historyDao);
+        hist = newDbHelper.getHistoryForDB("/test");
+        assertEquals(10, hist.size());
+        assertEquals(50.0, result.getMark());
```

Comment:
What is this assert checking?

Reply:
checks that its still 10 entries even after the 11th quiz and that the right object is being inserted.